tt_eager/tt_dnn/op_library/eltwise_binary/eltwise_binary_op.hpp

```diff
@@ -306,7 +306,7 @@ inline Tensor add(
                 EltwiseBinary{
                     BinaryOpType::ADD,
                     fused_activations,
-                    output_mem_config,
+                    output_mem_config.value(),
```

Comment:
Don't think we should be doing .value() here.

Is the plan to change the default value for output_mem_config to nullopt instead of operation::DEFAULT_OUTPUT_MEMORY_CONFIG?

I think here we should either:

- pass output_mem_config as is or
- pass output_mem_config.value_or(whatever we decide the default should be)

@eyonland do you have thoughts on this? Should we determine the default memconfig to use when none is provided here, or within the op struct fns?

Reply:
Yes, the default value of `output_mem_config` will be nullopt. but I am thinking of assigning it operation::DEFAULT_OUTPUT_MEMORY_CONFIG  **if**  both output_tensor and  output_mem_config is nullopt. 

I have an earlier comment from @eyonland regarding this, are you ok with this too ?

_> If the optional output tensor is provided, then it should be used instead of the dtype or output memconfig which should not be required but made optional going forward.
> The flow should be:
> Does an optional output tensor actually exist that is NOT a promise?  If so, ignore whatever the user provided for dtype and memconfig.
> If dtype and memconfig are provided use this, if not use the dtype and memconfig of the first input tensor as the default(s)_

so I'm thinking of adding an if block inside the `run_eltwise_binary` and change `output_mem_config` as per above conditions and then pass it to the `EltwiseBinary` struct